tutorials/raw/plot_object_raw.py

```diff
@@ -2,131 +2,557 @@
+# +-------------------------------------+                         |           |
+# | ``raw[[2, 5], 1000:2000]``          | :class:`tuple` of       |           |
+# +-------------------------------------+ (data (2 × 1000),       | no        |
+# | ``raw[['EEG 030', 'EOG 061'],       | times (1 × 1000))       |           |
+# | 1000:2000]``                        |                         |           |
+# +-------------------------------------+-------------------------+-----------+
+# | ``raw.pick_types(meg=False,         | :class:`~mne.io.Raw`    |           |
+# | eeg=True)``                         | object w/ only EEG      | yes       |
+# |                                     | channels                |           |
+# +-------------------------------------+-------------------------+-----------+
+#
+#
+# Renaming channels
+# ^^^^^^^^^^^^^^^^^
+#
+# .. sidebar:: Long channel names
+#
+#     Due to limitations in the :file:`.fif` file format (which MNE-Python uses
+#     to save :class:`~mne.io.Raw` objects), channel names are limited to a
+#     maximum of 15 characters.
+#
+# You may have noticed that the EEG channel names in the sample data are
+# numbered rather than labelled according to a standard nomenclature such as
+# the `10-20 <ten_twenty>`_ or `10-05 <ten_oh_five>`_ systems, or perhaps it
+# bothers you that the channel names contain spaces. It is possible to rename
+# channels using the :meth:`~mne.io.Raw.rename_channels` method, which takes a
+# Python dictionary to map old names to new names. You need not rename all
+# channels at once; provide only the dictionary entries for the channels you
+# want to rename. Here's a frivolous example:
+
+raw.rename_channels({'EOG 061': 'blink detector'})
+
+###############################################################################
+# This next example replaces spaces in the channel names with underscores,
+# using a Python `dict comprehension`_:
+
+print(raw.ch_names[-3:])
+channel_renaming_dict = {name: name.replace(' ', '_') for name in raw.ch_names}
+raw.rename_channels(channel_renaming_dict)
+print(raw.ch_names[-3:])
+
+###############################################################################
+# Changing channel type
+# ^^^^^^^^^^^^^^^^^^^^^
+#
+# If for some reason the channel types in your :class:`~mne.io.Raw` object are
+# inaccurate, you can change the type of any channel with the
+# :meth:`~mne.io.Raw.set_channel_types` method. The method takes a
+# :class:`dictionary <dict>` mapping channel names to types; allowed types are
+# ``ecg, eeg, emg, eog, exci, ias, misc, resp, seeg, stim, syst, ecog, hbo,
+# hbr``. A common use case for changing channel type is when using frontal EEG
+# electrodes as makeshift EOG channels:
+
+raw.set_channel_types({'EEG_001': 'eog'})
+print(raw.copy().pick_types(meg=False, eog=True).ch_names)
+
+###############################################################################
+# Cropping and concatenating Raw objects
```

Comment:
I think raw.pick_types belongs in a section with raw.crop and raw.append since what pick_types does is cropping but in another dimension

Reply:
Ideally I wanted the major split to be between "operations on channels" and "operations on the time domain", which is why `crop()` and `append()` ended up in a section together. But the "operations on channels" got messy because it seemed more efficient to talk about square-bracket-indexing both dimensions at the same time (and likewise for the `start` and `stop` args of `get_data()`).

It sounds like you're suggesting that the major split should be between "operations that return an array" and "operations that return a modified Raw object" (in which case it should also include `pick_channels()`, `drop_channels()`, and `reorder_channels()`. Is that in fact what you're suggesting?

Reply:
yes, that's my suggestion. Although the second category "operations that return a modified Raw object" would probably be "operation that modify the Raw object (and optionally return a copy) without actually modifying the data values". Probably not the best analogy but I like to call these cut/copy/paste operations whereas the indexing would be the equivalent of a selection.